src/modules/roc_sndio/target_sox/roc_sndio/sox_backend.cpp

```diff
@@ -187,7 +187,10 @@ SoxBackend::SoxBackend()
     sox_get_globals()->output_message_handler = log_handler;
 }
 
-void SoxBackend::set_frame_size(size_t size) {
+void SoxBackend::set_frame_size(core::nanoseconds_t frame_length,
+                                size_t sample_rate,
+                                packet::channel_mask_t channels) {
+    size_t size = packet::ns_to_size(frame_length, sample_rate, channels);
     core::Mutex::Lock lock(mutex_);
```

Comment:
Could you please reorder these 2 lines so that the lock will be the first? Unless required for performance reasons, we usually try to keep such locks in the first line of every method for readability.

Why: When you read code and see that some lines are placed before the lock, you have to check every time whether they access shared resources or not to ensure that's not a bug. And if the lock is always the first, you just don't need to think about it and so it's easier to read the code.

Reply:
done.